### ISSUE TITLE
Operator: Workspace

### ISSUE THREAD

**Issue by eparovyshnaya** [eclipse-passage/passage#948]
All the Operator functioning is spinning around creating and managing artifacts of the following types
  - metadata for products and customerst
  - product version keys
  - issued licenses

Currently we store them in home-dir with not much of a structure, which makes the long-term management and migrations complicated. 

The idea is to exploit Eclipse Workspace for the purpose. 

```
------------------------------------------------------------
Draft of the workspace structure is the following  
 - /features (*.features_xmi files to keep Feature Sets)
 - /products (*.products_xmi to keep Product Lines)
 - /users    (*.users_xmi to manage Customers and Users)
 - /plans    (*.licenses_xmi to keep License Plans)
 - /keys     (storage for generated keys for Product Versions)
 -   /<product-id>
 -      /<product-version>
 -         <product-id>_<product-version>.pub
 -         <product-id>_<product-version>.scr
 - /issued   (all issued licenses)
 -   /<product-id>
 -      /<product-version>
 -         /<user-origin>
 -            /floating (floating liecnse packs)
 -               /*<floating-license-pack-id>
 -            /personal
 -               /<user-id>  (personal licenses)
 -                  *.lic
 -                  *.licen
               
------------------------------------------------------------
```

For a company, which uses Operator to issu licenses only for it's own product lines there can be only one Workspace of this structure. 

For a service centers, that supplies license issuing and management for another vendors' products, there should be one Workspace per vendor.

---
Version: 1.1.0 
Ported from [bugs.eclipse.org: 571523](https://bugs.eclipse.org/bugs/show_bug.cgi?id=571523)

**Comment by eparovyshnaya** [comment]
Ported comment [Alexander Fedorov 2021-04-09 09:11:09 EDT](https://bugs.eclipse.org/bugs/show_bug.cgi?id=571523#c1)
If we think about LDC it seems like the following structure should be created for passage-protected product:

```
com.acme.solution.installable/
    src/
    META-INF/
    OSGI-INF/
    com.acme.solution.installable.product
    plugin.xml
com.acme.solution.installable.dev/
    features/
    products/
com.acme.solution.installable.ops/
    keys/
    licenses/
    users/
```

**Comment by eparovyshnaya** [comment]
> Ported comment [Elena Parovyshnaia 2021-04-19 06:11:18 EDT](https://bugs.eclipse.org/bugs/show_bug.cgi?id=571523#c2)

As it's been found during discussions, all the data relative to a product licensing can be separated to three groups, each of which 
 - has a particular lifecycle
 - is managed by a used of distinct role, 
 - defines it's own rules of visibility and access, 
 - has particular level of sencibility and business value 
 - assumes distinguishable scenarios for data loss prevension

[1] 'definition': products and features are usually declared by development/ management, amended rarely and forms a ground for  ​
[2] 'evolution' data: product versions and feature versions (deliveries) managed by development, amended incrementally
[3] 'licening' artifacts: 
 - customer dase, 
 - encryption keys, 
 - license plans definition, 
 - actually issued license packs. 
In general case licensing activity 
 - is performed outside of development and dev management, 
 - deals with sencitive data (like maybe user personal information) and
 - business-valuable data (like customer base, product version keys, issued licenses). 
This part of a licensing project should be 
 - kept separately, 
 - available for a tight circle of operation officers
 - be maintained with regular secure backups

This separation should establish a base for the future product licensing workspace.

**Comment by eparovyshnaya** [comment]
> Ported comment [Elena Parovyshnaia 2021-04-22 06:15:28 EDT](https://bugs.eclipse.org/bugs/show_bug.cgi?id=571523#c3)

#750 Invite starting version of OperatorWorkspace interface and start implementing basing in the existing solutions

**Comment by eparovyshnaya** [comment]
Implemented to 2.3.0
